.github/workflows/reusable-build-docker-image.yaml

```diff
@@ -0,0 +1,225 @@
+name: Build docker image
+
+on:
+  # Reusable workflow
+  workflow_call:
+    secrets:
+      token:
+        required: true
+      docker_token:
+        required: true
+    outputs:
+      DOCKER_IMAGE_NAME:
+        description: "Name of docker image to be used"
+        value: ${{ jobs.get-job-flags.outputs.DOCLINTTEST_IMAGE }}
```

Comment:
We have to set any output variables we want to reference from the calling workflows here.